packages/server/src/api/controllers/view/viewsV2.ts

```diff
@@ -160,6 +164,57 @@ export async function create(ctx: Ctx<CreateViewRequest, CreateViewResponse>) {
   gridSocket?.emitViewUpdate(ctx, result)
 }
 
+async function handleViewFilterEvents(existingView: ViewV2, view: ViewV2) {
+  const filterGroups = view.queryUI?.groups?.length || 0
+  const properties = { filterGroups, tableId: view.tableId }
+  if (!existingView?.queryUI) {
+    await events.view.filterCreated(properties)
+  } else {
+    if (
+      filterGroups >
+      ((existingView && existingView?.queryUI?.groups?.length) || 0)
+    ) {
+      await events.view.filterUpdated(properties)
+    }
+  }
+}
+
+async function handleViewEvents(existingView: ViewV2, view: ViewV2) {
+  // Grouped filters
+  if (view.queryUI?.groups) {
+    await handleViewFilterEvents(existingView, view)
+  }
```

Comment:
This isn't a reliable indicator that grouped filters have been used. For backwards compatibility, we populate this `view.queryUI` field from `view.query` when `view.queryUI` hasn't been set: https://github.com/Budibase/budibase/blob/6865052ef0a64e8d5f43299d5d5f98ff8d829ce6/packages/server/src/sdk/app/views/utils.ts#L10-L32

In practice, this field will ~always be set.

Reply:
As far as I remember, these get run even on fetch. We wanted to make absolutely sure that these are always set, because the frontend chokes without them. I really wouldn't recommend relying on them.